inc/Engine/Common/JobManager/APIHandler/AbstractSafeAPIClient.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace WP_Rocket\Engine\Common\JobManager\APIHandler;
+
+use WP_Error;
+
+/**
+ * Class AbstractSafeAPIClient
+ *
+ * This abstract class provides a base for making API requests.
+ * It includes methods for sending GET and POST requests, and handles transient caching to mitigate the impact of API failures.
+ *
+ * @package WP_Rocket\Engine\Common\JobManager\APIHandler
+ */
+abstract class AbstractSafeAPIClient {
+
+	/**
+	 * Get the transient key.
+	 *
+	 * @return string The transient key.
+	 */
+	abstract protected function get_transient_key();
+
+	/**
+	 * Get the API URL.
+	 *
+	 * @return string The API URL.
+	 */
+	abstract protected function get_api_url();
+
+	/**
+	 * Send a GET request.
+	 *
+	 * @param array $params The request parameters.
+	 * @param bool  $safe Send safe request WP functions or not, default to not.
+	 *
+	 * @return mixed The response from the API.
+	 */
+	public function send_get_request( $params = [], $safe = false ) {
+		return $this->send_request( 'GET', $params, $safe );
+	}
+
+	/**
+	 * Send a POST request.
+	 *
+	 * @param array $params The request parameters.
+	 * @param bool  $safe Send safe request WP functions or not, default to not.
+	 *
+	 * @return mixed The response from the API.
+	 */
+	public function send_post_request( $params = [], $safe = false ) {
+		return $this->send_request( 'POST', $params, $safe );
+	}
+
+	/**
+	 * Send a request to the API.
+	 *
+	 * @param string $method The HTTP method (GET or POST).
+	 * @param array  $params The request parameters.
+	 * @param bool   $safe Send safe request WP functions or not, default to not.
+	 * @return mixed The response from the API, or WP_Error if a timeout is active.
+	 */
+	private function send_request( $method, $params = [], $safe = false ) {
+		$api_url = $this->get_api_url();
+
+		if ( get_transient( $this->get_transient_key() . '_timeout_active' ) ) {
+			return new WP_Error( 429, __( 'Too many requests.', 'rocket' ) );
```

Comment:
Maybe we could use a more specific string than the generic "Too many requests" one, at least to know the error is coming specifically from our transient system?
Or maybe put a log for debug mode?

Reply:
To do later on